src/table/use-sticky-columns.tsx

```diff
@@ -0,0 +1,224 @@
+  const lastStickyColumnsWidth = cellOffsets.last[lastRightStickyColumnIndex] ?? 0;
+
+  // Calculate the sum of all sticky columns' widths
+  const totalStickySpace = firstStickyColumnsWidth + lastStickyColumnsWidth;
+
+  useEffect(() => {
+    // Check if there is enough scrollable space for sticky columns to be enabled
+    const hasEnoughScrollableSpace =
+      totalStickySpace + MINIMUM_SCROLLABLE_SPACE + tableLeftPadding < (containerWidth ?? 0);
+
+    // Determine if sticky columns should be disabled based on the conditions
+    const shouldDisable = noStickyColumns || !isWrapperScrollable || !hasEnoughScrollableSpace;
+    setShouldDisable(shouldDisable);
+  }, [
+    containerWidth,
+    noStickyColumns,
+    stickyColumns,
+    totalStickySpace,
+    visibleColumnsLength,
+    tableLeftPadding,
+    isWrapperScrollable,
+  ]);
+
+  useEffect(() => {
+    // Create new refs for the visible columns and selection column, if present
+    setTableCellRefs(tableCellRefs =>
+      [...new Array(columnsLengthWithSelection)].map(
+        (_: any, i: number) => tableCellRefs[i] || createRef<HTMLTableCellElement>()
+      )
+    );
+  }, [columnsLengthWithSelection, hasSelection]);
+
+  const getStickyStyles = useCallback(
+    (colIndex: number, stickySide: 'left' | 'right' | '') => {
+      if (!stickySide) {
+        return { sticky: {}, stuck: {} };
+      }
+      const isFirstColumn = colIndex === 0;
+      let stuck = {};
+      if (isFirstColumn && !hasSelection && tableLeftPadding !== 0) {
+        stuck = { paddingLeft: `${tableLeftPadding}px` };
+      }
+      // Determine the offset of the sticky column using the `cellOffsets` state object
+      const offsetKey = stickySide === 'right' ? 'last' : 'first';
+      const stickyColumnOffset = cellOffsets[offsetKey]?.[colIndex + (hasSelection ? 1 : 0)];
+      return {
+        sticky: typeof stickyColumnOffset !== 'undefined' ? { [stickySide]: `${stickyColumnOffset}px` } : {},
+        stuck,
+      };
+    },
+    [cellOffsets, hasSelection, tableLeftPadding]
+  );
+
+  const getStickyColumnProperties = React.useCallback(
+    (colIndex: number): GetStickyColumnProperties => {
+      const disabledStickyColumn = {
+        isSticky: false,
+        isLastStickyLeft: false,
+        isLastStickyRight: false,
+        stickyStyles: { sticky: {}, stuck: {} },
+      };
+
+      if (shouldDisable) {
+        return disabledStickyColumn;
+      }
+
+      // Determine if the column is sticky on the left or right side
+      const isStickyLeft = colIndex + 1 <= (stickyColumns?.first ?? 0);
+      const isStickyRight = colIndex + 1 > visibleColumnsLength - (stickyColumns?.last ?? 0);
+
+      // Determine if the column is the last left or right sticky column
+      const isLastStickyLeft = colIndex + 1 === stickyColumns?.first;
+      const isLastStickyRight = colIndex === visibleColumnsLength - (stickyColumns?.last ?? 0);
```

Comment:
We can possibly encapsulate this somewhere. E.g. `useCellOffsets` or other hook can return a structure with helpers like `isSticky(columnIndex: number): boolean`, `isLastSticky(columnIndex: number)` etc.

Like:
```
interface StickyState {
  totalVisibleColumns: number;
  getLeftOffset: (index: number) => number;
  isStickyLeft: (index: number) => number;
  ... 
}
```

Reply:
Tackling larger hook refactoring post-release.